mlir/lib/Dialect/Affine/Transforms/AffineLoopInterchange.cpp

```diff
@@ -0,0 +1,964 @@
+/// these to protected members `loopAccessMatrices`(A) and `constVector`(B).
+/// The param `indVarIdMap` is used to locate position for coefficients of X
+/// in each row of matrix A.
+void LoopInterchange::getAffineAccessMatrices(
+    AffineForOp &rootForOp, SmallVector<Operation *, 8> &loadAndStoreOps,
+    DenseMap<Value, unsigned> &indVarIdMap, unsigned AffineForOpLoopNestSize) {
+
+  this->loopAccessMatrices.clear();
+  this->constVector.clear();
+
+  unsigned numOps = loadAndStoreOps.size();
+  for (unsigned i = 0; (i < numOps); ++i) {
+    Operation *srcOp = loadAndStoreOps[i];
+    MemRefAccess srcAccess(srcOp);
+
+    AffineMap map;
+    if (AffineLoadOp loadOp = dyn_cast<AffineLoadOp>(srcAccess.opInst))
+      map = loadOp.getAffineMap();
+    else if (AffineStoreOp storeOp = dyn_cast<AffineStoreOp>(srcAccess.opInst))
+      map = storeOp.getAffineMap();
+
+    SmallVector<Value, 8> operands(srcAccess.indices.begin(),
+                                   srcAccess.indices.end());
+
+    fullyComposeAffineMapAndOperands(&map, &operands);
+    map = simplifyAffineMap(map);
+    canonicalizeMapAndOperands(&map, &operands);
+
+    // Parse each map result to construct access matrices.
+    ArrayRef<AffineExpr> mapResults = map.getResults();
+
+    uint64_t noDims = map.getNumDims();
+    uint64_t noSymbols = map.getNumSymbols();
+
+    // Number of rows in an accessMatrix = Number of dimensions
+    // in the memref object.
+    // Number of Columns = (noDims + noSymbols).
+
+    SmallVector<SmallVector<int64_t, 4>, 4> accessMatrix;
+    SmallVector<int64_t, 4> constVector(mapResults.size());
+
+    for (unsigned l = 0; l < mapResults.size(); l++) {
+
+      // Parse the l-th map result (access expr for l-th dimension of this
+      // memref) to get the l-th row of this access matrix.
+      AffineExpr mapResult = mapResults[l];
+
+      // A Row of Matrix A.
+      SmallVector<int64_t, 4> Row;
+      Row.resize(
+          std::max(AffineForOpLoopNestSize, unsigned(noDims + noSymbols)));
+
+      // Check if `mapResult` is not a constant expr.
+      // If it is a constant expr like A[5], no need to walk it.
+      // Instead, push the value in constVector and
+      // leave the Row matrix to be a vector of zeroes.
+
+      if (mapResult.getKind() == AffineExprKind::Constant) {
+        auto constant = mapResult.cast<AffineConstantExpr>();
+        constVector[l] = (constant.getValue());
+      } else {
+
+        // Start parsing the mapResult.
+        mapResult.walk([&](AffineExpr expr) {
+          // Each `expr` can in turn be a combination of other affine
+          // expressions.
+          switch (expr.getKind()) {
+          case AffineExprKind::Add: {
+
+            // Does this row need a modification?
+            // Please note that the Row should not
+            // be modified in case of a constant affine expr.
+            bool modifyRow = true;
```

Comment:
modifyRow or rowModified?

Reply:
Whether this row will be modified or not (which initially is full of zeros) is a future decision based on the value of this variable. Using rowModified gives a sense of past as if the row has been modified. Hence I think modifyRow is more suitable?

Reply:
Replaced the variable name 'modifyRow' with 'isConstExpr'. It is more suitable since at last all we are doing here is to test whether a sub-expr is constant or not. In case of a constant, we would not modify the row.